src/mongo/jobs/EnsureIndexes.class.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Tripod\Mongo\Jobs;
+
+/**
+ * Class EnsureIndexes
+ * @package Tripod\Mongo\Jobs
+ */
+class EnsureIndexes extends JobBase {
+
+    const STORENAME_KEY  = 'storeName';
+    const REINDEX_KEY    = 'reindex';
+
+    /**
+     * Runs the EnsureIndexes Job
+     * @throws \Exception
+     */
+    public function perform()
+    {
+        try
+        {
+            $this->debugLog("[JOBID " . $this->job->payload['id'] . "] EnsureIndexes::perform() start");
+
+            $timer = new \Tripod\Timer();
+            $timer->start();
+
+            $this->validateArgs();
+
+            \Tripod\Mongo\Config::setConfig($this->args[self::TRIPOD_CONFIG_KEY]);
+
+            $this->getIndexUtils()->ensureIndexes(
+                $this->args[self::REINDEX_KEY],
+                $this->args[self::STORENAME_KEY],
+                true // always create indexes in the background
```

Comment:
We support them being created in the background/foreground in the ensureIndexes function - maybe we should support creating them in the job, even if we default back to background. Or maybe hard coding it to `true` here is good enough as we may always want to create it in the BG.